### ISSUE TITLE
Track artist is not displayed in browse results

### ISSUE THREAD

**Issue by kingosticks** [pimusicbox/mopidy-musicbox-webclient#99]
Originally posted at https://discuss.mopidy.com/t/artist-name-not-displayed-in-several-screens/682

>I'm just in the process of upgrading from musicbox 0.53 to 0.6 and noticed that the artist name is no longer displayed in the "Queue" screen. 
While I quite like the additional line showing the album name and album artist, the individual artists of the tracks are missing. This makes identifying the songs rather difficult.

**Comment by kingosticks** [comment]
On further inspection, it seems that in the playlist and search pages, if there is just a single track from an album (and therefore no grey album row) you are shown the artist name alongside the track name. If you play from these pages then this is displayed the same way in the queue. So this is  intentional behaviour. 

Browse works somewhat differently.  Here we are returned lightweight Ref objects which don't contain the artist/album metadata. So you only get the track name. But since Mopidy v1.0+ is moving towards using Refs everywhere we'll need to handle this anyway. I guess we need to do lookups on all the returned results...

**Comment by jcass77** [comment]
I have a question on this, having recently worked on refactoring the code responsible for displaying list of tracks as part of the search results and the 'now playing' queue.

Although I appreciate that it would be nice to have all of the various lists look and behave in an identical way, is the intention of using Refs not specifically to improve performance? I did a quick test and doing lookups on large numbers of track Refs does cause a noticeable delay during which the browser becomes unresponsive, even when using promises.

Is an effective implementation of this not blocked by the ability to paginate results as described in https://github.com/mopidy/mopidy/issues/1392 and https://github.com/mopidy/mopidy/issues/733?

In the interim, using something like [mopidy-local-sqlite](https://github.com/mopidy/mopidy-local-sqlite) might be a more effective means of browsing large collections of albums and artists.

**Comment by kingosticks** [comment]
Yes, `Refs` are a big improvement when you can save transferring lots of metadata around *that you don't care about*. I think the idea here was saying that maybe we do care about this rich metadata when browsing. You could argue that for the `local` backend, we do/should care but for the `file` backend we don't. 

I'm not sure I understand how the user's particular local backend affects us.
